### PR TITLE
Finished CCam; various smaller things

### PR DESCRIPTION
**Pull request by aap**

### CODE REVIEW

File: src/core/Cam.cpp [hunk 1]
```diff
@@ -0,0 +1,4150 @@
...
+
+	if((Mode == MODE_CAM_ON_A_STRING || Mode == MODE_BEHINDBOAT) && CamTargetEntity->IsVehicle()){
+		LookingBehind = true;
+		Dist = Mode == MODE_CAM_ON_A_STRING ? CA_MAX_DISTANCE : 15.5f;
+		TargetFwd = CamTargetEntity->GetForward();
+		TargetFwd.Normalise();
+		TargetOrientation = CGeneral::GetATanOfXY(TargetFwd.x, TargetFwd.y);
+		DeltaBeta = TargetOrientation - Beta;
+		while(DeltaBeta >= PI) DeltaBeta -= 2*PI;
+		while(DeltaBeta < -PI) DeltaBeta += 2*PI;
```
Comment (erorcun) [inline src/core/Cam.cpp]: there are many instances of that in the file, and they all looks like MakeAngleLessThan180

File: src/core/Cam.cpp [hunk 2]
```diff
@@ -0,0 +1,4150 @@
...
+	}
+
+	bool OnTrain = FindPlayerVehicle() && FindPlayerVehicle()->IsTrain();
+
+	// Look around
+	bool UseMouse = false;
+	float MouseX = CPad::GetPad(0)->GetMouseX();
+	float MouseY = CPad::GetPad(0)->GetMouseY();
+	float LookLeftRight, LookUpDown;
+	if((MouseX != 0.0f || MouseY != 0.0f) && !CPad::GetPad(0)->DisablePlayerControls){
```
Comment (erorcun) [inline src/core/Cam.cpp]: !ArePlayerControlsDisabled()

File: src/core/Cam.cpp [hunk 3]
```diff
@@ -0,0 +1,4150 @@
...
+
+	Source.z = TargetCoors.z + Sin(Alpha + ModeAlpha)*Length + m_fCloseInCarHeightOffset;
+}
+
+// Rotate cam behind the car when the car is moving forward
+bool
+CCam::RotCamIfInFrontCar(CVector &TargetCoors, float TargetOrientation)
+{
+	bool MovingForward = false;
+	CPhysical *p
```
Comment (erorcun) [inline src/core/Cam.cpp]: ped->GetNodeFrame

Comment (erorcun) [inline src/core/Cam.cpp]: process_process_ ? :D

Comment (erorcun) [inline src/core/Cam.cpp]: ped->GetNodeFrame

Comment (erorcun) [inline src/core/Cam.cpp]: process_process_ ? :D

File: src/core/Cam.cpp (hunk 1 above)
Comment (aap) [reply src/core/Cam.cpp]: i'll leave it for now

File: src/core/Cam.cpp [hunk 4]
```diff
@@ -0,0 +1,4147 @@
...
+
+	if(TheCamera.m_bcutsceneFinished)
+		return;
+
+	Up = CVector(0.0f, 0.0f, 1.0f);
+	if(TheCamera.m_bStartingSpline)
+		m_fTimeElapsedFloat += CTimer::GetTimeStepInMilliseconds();
+	else{
+		m_fTimeElapsedFloat = 0.0f;
+		m_uiFinishTime = MS(TheCamera.m_arrPathArray[2].m_arr_PathData[10*((int)TheCamera.m_arrPathArray[2].m_arr_PathData[0]-1) + 1]);
```
Comment (ShFil119) [inline src/core/Cam.cpp]: int32

File: src/core/Cam.cpp [hunk 5]
```diff
@@ -0,0 +1,4147 @@
...
+}
+
+// unused
+void
+CCam::Process_Circle(const CVector &CameraTarget, float, float, float)
+{
+	FOV = DefaultFOV;
+
+	Front.x =  Cos(0.7f) * Cos((CTimer::GetTimeInMilliseconds()&0xFFF)/(float)0xFFF * TWOPI);
+	Front.y =  Cos(0.7f) * Sin((CTimer::GetTimeInMilliseconds()&0xFFF)/(float)0xFFF * TWOPI);
```
Comment (ShFil119) [inline src/core/Cam.cpp]: decimal?

File: src/core/Camera.h [hunk 6]
```diff
@@ -445,8 +474,8 @@ uint32    unknown;
 uint32  m_fScriptTimeForInterPolation;
 
 
-int16   m_iFadingDirection;
-int     m_iModeObbeCamIsInForCar;
+	int16   m_iFadingDirection;
+	int     m_iModeObbeCamIsInForCar;
```
Comment (ShFil119) [inline src/core/Camera.h]: int32

File: src/core/Cam.cpp (hunk 4 above)
Comment (aap) [reply src/core/Cam.cpp]: int is fine in this case.

File: src/core/Cam.cpp (hunk 5 above)
Comment (aap) [reply src/core/Cam.cpp]: i find hex more readable

File: src/core/Camera.h (hunk 6 above)
Comment (aap) [reply src/core/Camera.h]: CCamera has a lot of ints, and it comes from R*, so i think we'll leave it.

**Pull request merged**